### ISSUE TITLE
yue module contains references to all content of package.loaded

### ISSUE THREAD

**Issue by johnd0e** [pigpigyyy/Yuescript#132]
Just found really strange issue in latest release:
~~~
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2021 Mike Pall. https://luajit.org/
JIT: ON SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> yue = require "yue"
> print(yue.version, yue.yue, yue.yue.yue, yue.string, yue.table, yue._G)
0.16.0  table: 0x00692588       table: 0x00692588       table: 0x00695058       table: 0x00693860       table: 0x006913c0
>
~~~

Compare with previous build:
~~~
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2021 Mike Pall. https://luajit.org/
JIT: ON SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> yue = require "yue"
> print(yue.version, yue.yue, yue.string, yue.table, yue._G)
0.15.29 nil     nil     nil     nil
>
~~~

**Comment by johnd0e** [comment]
And vice versa:
```
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2021 Mike Pall. https://luajit.org/
JIT: ON SSE3 SSE4.1 BMI2 fold cse dce fwd dse narrow loop abc sink fuse
> yue = require"yue"
> print(package.loaded.to_ast)
function: 0x001d8838
```

**Comment by pigpigyyy** [comment]
Found the problem to be a misuse of luaL_register() from Lua 5.1 C API. Got the bug fixed by commit a8c4737494818e6591cac2191bf3a49cbe990173
